ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFStack.java

```diff
@@ -63,11 +64,16 @@ public StructObjectInspector initialize(ObjectInspector[] args)
     }
     if (!(args[0] instanceof ConstantObjectInspector)) {
       throw new UDFArgumentException(
+          "The first argument to STACK() must be a constant.");
+    }
+    final Object value = ((ConstantObjectInspector) args[0]).getWritableConstantValue();
+    if (!(value instanceof IntWritable)) {
```

Comment:
I added a validation since the current implementation throws a ClassCastException. I can split PRs if we don't want to fix it in this PR.